Container.lua

```diff
@@ -128,19 +128,21 @@ end
 
 function Container:clearState()
    -- don't call set because it might reset referenced tensors
-   local function clear(f)
-      if self[f] then
-         if torch.isTensor(self[f]) then
-            self[f] = self[f].new()
-         elseif type(self[f]) == 'table' then
-            self[f] = {}
-         else
-            self[f] = nil
+   local function clear(t)
+      if torch.isTensor(t) then
+         return t.new()
```

Comment:
I think that if https://github.com/torch/nn/pull/673 gets merged, we will be able to just call `:set()` on the tensors and keep all the references to the network tensors after `clearState`.